<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>814</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>ESP8266 Helper GUI</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QGroupBox" name="Firmware">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>791</width>
      <height>231</height>
     </rect>
    </property>
    <property name="title">
     <string>Upload Firmware</string>
    </property>
    <widget class="QPushButton" name="btnFirmRebuild">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>200</y>
       <width>121</width>
       <height>27</height>
      </rect>
     </property>
     <property name="text">
      <string>Re-Build Firmware</string>
     </property>
    </widget>
    <widget class="QGroupBox" name="FirmPath">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>30</y>
       <width>421</width>
       <height>151</height>
      </rect>
     </property>
     <property name="title">
      <string>Firmware Path</string>
     </property>
     <widget class="QLineEdit" name="txtFirm">
      <property name="geometry">
       <rect>
        <x>80</x>
        <y>60</y>
        <width>321</width>
        <height>24</height>
       </rect>
      </property>
      <property name="readOnly">
       <bool>true</bool>
      </property>
     </widget>
     <widget class="QLabel" name="lblPath">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>30</y>
        <width>61</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string>Workspace</string>
      </property>
     </widget>
     <widget class="QLabel" name="lblIBlankBin">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>120</y>
        <width>70</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string>Blank bin</string>
      </property>
     </widget>
     <widget class="QLabel" name="lblInitBin">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>90</y>
        <width>70</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string>Init bin</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="txtBlankBin">
      <property name="geometry">
       <rect>
        <x>80</x>
        <y>120</y>
        <width>321</width>
        <height>24</height>
       </rect>
      </property>
      <property name="readOnly">
       <bool>true</bool>
      </property>
     </widget>
     <widget class="QLineEdit" name="txtPath">
      <property name="geometry">
       <rect>
        <x>80</x>
        <y>30</y>
        <width>321</width>
        <height>24</height>
       </rect>
      </property>
      <property name="readOnly">
       <bool>true</bool>
      </property>
     </widget>
     <widget class="QLabel" name="lblFirm">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>60</y>
        <width>61</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string>Firmware</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="txtInitBin">
      <property name="geometry">
       <rect>
        <x>80</x>
        <y>90</y>
        <width>321</width>
        <height>24</height>
       </rect>
      </property>
      <property name="readOnly">
       <bool>true</bool>
      </property>
     </widget>
    </widget>
    <widget class="QGroupBox" name="FirmAddress">
     <property name="geometry">
      <rect>
       <x>450</x>
       <y>30</y>
       <width>171</width>
       <height>121</height>
      </rect>
     </property>
     <property name="title">
      <string>Binary Address</string>
     </property>
     <widget class="QLineEdit" name="txtAddrBlank1">
      <property name="geometry">
       <rect>
        <x>70</x>
        <y>20</y>
        <width>71</width>
        <height>24</height>
       </rect>
      </property>
     </widget>
     <widget class="QLabel" name="lblAddrBlank1">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>20</y>
        <width>54</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string>1st Blank</string>
      </property>
     </widget>
     <widget class="QLabel" name="lblAddrInit">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>50</y>
        <width>54</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string>Init Data</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="txtAddrInit">
      <property name="geometry">
       <rect>
        <x>70</x>
        <y>50</y>
        <width>71</width>
        <height>24</height>
       </rect>
      </property>
     </widget>
     <widget class="QLineEdit" name="txtAddrBlank2">
      <property name="geometry">
       <rect>
        <x>70</x>
        <y>80</y>
        <width>71</width>
        <height>24</height>
       </rect>
      </property>
     </widget>
     <widget class="QLabel" name="lblAddrBlank2">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>80</y>
        <width>54</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string>2nd Blank</string>
      </property>
     </widget>
    </widget>
    <widget class="QGroupBox" name="FirmUser">
     <property name="geometry">
      <rect>
       <x>630</x>
       <y>30</y>
       <width>151</width>
       <height>121</height>
      </rect>
     </property>
     <property name="title">
      <string>User Binary Address</string>
     </property>
     <widget class="QLabel" name="lblAddrUser1">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>20</y>
        <width>54</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string>1st Part</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="txtAddrUser1">
      <property name="geometry">
       <rect>
        <x>70</x>
        <y>20</y>
        <width>71</width>
        <height>24</height>
       </rect>
      </property>
      <property name="readOnly">
       <bool>true</bool>
      </property>
     </widget>
     <widget class="QLabel" name="lblAddrUser2">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>50</y>
        <width>54</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string>2nd Part</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="txtAddrUser2">
      <property name="geometry">
       <rect>
        <x>70</x>
        <y>50</y>
        <width>71</width>
        <height>24</height>
       </rect>
      </property>
      <property name="readOnly">
       <bool>true</bool>
      </property>
     </widget>
    </widget>
    <widget class="QGroupBox" name="ESPtool">
     <property name="geometry">
      <rect>
       <x>500</x>
       <y>150</y>
       <width>281</width>
       <height>80</height>
      </rect>
     </property>
     <property name="title">
      <string>ESPtool</string>
     </property>
     <widget class="QLabel" name="lblESProot">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>30</y>
        <width>81</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Rooting</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="txtESPport">
      <property name="geometry">
       <rect>
        <x>190</x>
        <y>50</y>
        <width>81</width>
        <height>24</height>
       </rect>
      </property>
     </widget>
     <widget class="QLineEdit" name="txtESPtool">
      <property name="geometry">
       <rect>
        <x>100</x>
        <y>50</y>
        <width>81</width>
        <height>24</height>
       </rect>
      </property>
     </widget>
     <widget class="QLabel" name="lblESPtool">
      <property name="geometry">
       <rect>
        <x>100</x>
        <y>30</y>
        <width>81</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Script</string>
      </property>
     </widget>
     <widget class="QLabel" name="lblESPport">
      <property name="geometry">
       <rect>
        <x>190</x>
        <y>30</y>
        <width>71</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Device Port</string>
      </property>
     </widget>
     <widget class="QComboBox" name="cmbESProot">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>50</y>
        <width>78</width>
        <height>24</height>
       </rect>
      </property>
      <property name="currentIndex">
       <number>1</number>
      </property>
      <item>
       <property name="text">
        <string>none</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>sudo</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>su</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>gksudo</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>gksu</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>kdesudo</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>kdesu</string>
       </property>
      </item>
     </widget>
    </widget>
    <widget class="QPushButton" name="btnFirmUpload">
     <property name="geometry">
      <rect>
       <x>150</x>
       <y>200</y>
       <width>121</width>
       <height>27</height>
      </rect>
     </property>
     <property name="text">
      <string>Upload Firmware</string>
     </property>
    </widget>
    <widget class="QPushButton" name="btnFirmClean">
     <property name="geometry">
      <rect>
       <x>290</x>
       <y>200</y>
       <width>121</width>
       <height>27</height>
      </rect>
     </property>
     <property name="text">
      <string>Clean-Up Firmware</string>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="DeviceID">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>320</y>
      <width>421</width>
      <height>251</height>
     </rect>
    </property>
    <property name="title">
     <string>Device ID</string>
    </property>
    <widget class="QLineEdit" name="txtServReq">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>40</y>
       <width>291</width>
       <height>23</height>
      </rect>
     </property>
     <property name="readOnly">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QLabel" name="lblServReq">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>20</y>
       <width>111</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Server Request URL</string>
     </property>
    </widget>
    <widget class="QPushButton" name="btnServReq">
     <property name="geometry">
      <rect>
       <x>310</x>
       <y>40</y>
       <width>85</width>
       <height>27</height>
      </rect>
     </property>
     <property name="text">
      <string>Request</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="txtDeviceID">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>100</y>
       <width>291</width>
       <height>23</height>
      </rect>
     </property>
     <property name="readOnly">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QPushButton" name="btnSetDevID">
     <property name="geometry">
      <rect>
       <x>310</x>
       <y>100</y>
       <width>85</width>
       <height>27</height>
      </rect>
     </property>
     <property name="text">
      <string>Set DevID</string>
     </property>
    </widget>
    <widget class="QLabel" name="lblSetDevID">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>80</y>
       <width>131</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Device ID from Server</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="txtConfirm">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>220</y>
       <width>291</width>
       <height>23</height>
      </rect>
     </property>
     <property name="readOnly">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QPushButton" name="btnConfirm">
     <property name="geometry">
      <rect>
       <x>310</x>
       <y>220</y>
       <width>85</width>
       <height>27</height>
      </rect>
     </property>
     <property name="text">
      <string>Confirm</string>
     </property>
    </widget>
    <widget class="QLabel" name="lblConfirm">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>200</y>
       <width>171</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Device ID confirm URL</string>
     </property>
    </widget>
    <widget class="QLabel" name="lblChkDevID">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>140</y>
       <width>171</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Device ID check result </string>
     </property>
    </widget>
    <widget class="QLineEdit" name="txtChkDevID">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>160</y>
       <width>291</width>
       <height>23</height>
      </rect>
     </property>
     <property name="readOnly">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QPushButton" name="btnChkDevID">
     <property name="geometry">
      <rect>
       <x>310</x>
       <y>160</y>
       <width>85</width>
       <height>27</height>
      </rect>
     </property>
     <property name="text">
      <string>Check</string>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="Serial">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>250</y>
      <width>421</width>
      <height>61</height>
     </rect>
    </property>
    <property name="title">
     <string>Serial UART</string>
    </property>
    <widget class="QLabel" name="lblBaud">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>30</y>
       <width>51</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>Baudrate</string>
     </property>
    </widget>
    <widget class="QComboBox" name="cmbBaud">
     <property name="geometry">
      <rect>
       <x>80</x>
       <y>30</y>
       <width>111</width>
       <height>24</height>
      </rect>
     </property>
     <item>
      <property name="text">
       <string>9600</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>19200</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>38400</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>57600</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>115200</string>
      </property>
     </item>
    </widget>
    <widget class="QPushButton" name="btnConnect">
     <property name="geometry">
      <rect>
       <x>210</x>
       <y>30</y>
       <width>91</width>
       <height>27</height>
      </rect>
     </property>
     <property name="text">
      <string>Connect</string>
     </property>
    </widget>
    <widget class="QPushButton" name="btnClear">
     <property name="geometry">
      <rect>
       <x>320</x>
       <y>30</y>
       <width>85</width>
       <height>27</height>
      </rect>
     </property>
     <property name="text">
      <string>Clear</string>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="Console">
    <property name="geometry">
     <rect>
      <x>440</x>
      <y>250</y>
      <width>361</width>
      <height>321</height>
     </rect>
    </property>
    <property name="title">
     <string>Serial Console</string>
    </property>
    <widget class="QPushButton" name="btnInfoJSON">
     <property name="geometry">
      <rect>
       <x>100</x>
       <y>30</y>
       <width>61</width>
       <height>27</height>
      </rect>
     </property>
     <property name="text">
      <string>JSON Info</string>
     </property>
    </widget>
    <widget class="QPushButton" name="btnRestart">
     <property name="geometry">
      <rect>
       <x>170</x>
       <y>30</y>
       <width>51</width>
       <height>27</height>
      </rect>
     </property>
     <property name="text">
      <string>Restart</string>
     </property>
    </widget>
    <widget class="QPushButton" name="btnSwitch">
     <property name="geometry">
      <rect>
       <x>230</x>
       <y>30</y>
       <width>41</width>
       <height>27</height>
      </rect>
     </property>
     <property name="text">
      <string>Switch</string>
     </property>
    </widget>
    <widget class="QPushButton" name="btnInfoSystem">
     <property name="geometry">
      <rect>
       <x>24</x>
       <y>30</y>
       <width>71</width>
       <height>27</height>
      </rect>
     </property>
     <property name="text">
      <string>System Info</string>
     </property>
    </widget>
    <widget class="QPlainTextEdit" name="txtConsole">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>70</y>
       <width>341</width>
       <height>241</height>
      </rect>
     </property>
     <property name="readOnly">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QPushButton" name="btnRequest">
     <property name="geometry">
      <rect>
       <x>280</x>
       <y>30</y>
       <width>51</width>
       <height>27</height>
      </rect>
     </property>
     <property name="text">
      <string>Request</string>
     </property>
    </widget>
   </widget>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
